src/lib/datepicker/calendar-body.ts

```diff
@@ -175,16 +175,18 @@ export class MatCalendarBody<D = unknown> implements OnChanges, OnDestroy {
   }
 
   _cellClicked(cell: MatCalendarCell<D>): void {
-    if (cell.enabled && this._selectionModel instanceof MatSingleDateSelectionModel) {
-      const date = cell.range.start;
-      const granularity = this._getFirstCellGranularity();
-      if (granularity == 'year') {
-        this.selectedValueChange.emit(this._dateAdapter.getYear(date));
-      } else if (granularity == 'month') {
-        this.selectedValueChange.emit(this._dateAdapter.getMonth(date));
-      } else {
-        this.selectedValueChange.emit(this._dateAdapter.getDate(date));
-      }
+    if (!cell.enabled) {
```

Comment:
Not sure why we ever checked which sub-type the `dateSelection` is but it seems to make no difference.